nemo/collections/asr/models/hybrid_asr_tts_models.py

```diff
@@ -0,0 +1,459 @@
+        if name == "_current_fx_name" and self._full_init_guard:
+            self.asr_model._current_fx_name = value  # need to make logging inside asr_model work
+        return super().__setattr__(name, value)
+
+    def setup_optimization(
+        self, optim_config: Optional[Union[DictConfig, Dict]] = None, optim_kwargs: Optional[Dict[str, Any]] = None,
+    ):
+        self.tts_model.freeze()
+        optimizer, scheduler = super().setup_optimization(optim_config=optim_config, optim_kwargs=optim_kwargs)
+        # set ASR model optimizer/scheduler to allow training_step on asr_model
+        self.asr_model._optimizer = optimizer
+        self.asr_model._scheduler = scheduler
+        return optimizer, scheduler
+
+    def setup_validation_data(self, val_data_config: Union[DictConfig, Dict]):
+        return self.asr_model.setup_validation_data(val_data_config)
+
+    def multi_validation_epoch_end(self, outputs, dataloader_idx: int = 0):
+        return self.asr_model.multi_validation_epoch_end(outputs=outputs, dataloader_idx=dataloader_idx)
+
+    def multi_test_epoch_end(self, outputs, dataloader_idx: int = 0):
+        return self.asr_model.multi_test_epoch_end(outputs=outputs, dataloader_idx=dataloader_idx)
+
+    def transcribe(self, paths2audio_files: List[str], batch_size: int = 4) -> List[str]:
+        return self.asr_model.transcribe(paths2audio_files=paths2audio_files, batch_size=batch_size)
+
+    def setup_multiple_validation_data(self, val_data_config: Union[DictConfig, Dict]):
+        self.asr_model.setup_multiple_validation_data(val_data_config)
+
+    def setup_test_data(self, test_data_config: Union[DictConfig, Dict]):
+        self.asr_model.setup_test_data(test_data_config)
+
+    def setup_multiple_test_data(self, test_data_config: Union[DictConfig, Dict]):
+        return self.asr_model.setup_multiple_test_data(test_data_config)
+
+    def save_asr_model_to(self, save_path: str):
+        return self.asr_model.save_to(save_path=save_path)
+
+    def validation_step(self, batch, batch_idx, dataloader_idx=0):
+        return self.asr_model.validation_step(batch=batch, batch_idx=batch_idx, dataloader_idx=dataloader_idx)
+
+    def validation_epoch_end(
+        self, outputs: Union[List[Dict[str, torch.Tensor]], List[List[Dict[str, torch.Tensor]]]]
+    ) -> Optional[Dict[str, Dict[str, torch.Tensor]]]:
+        return self.asr_model.validation_epoch_end(outputs=outputs)
+
+    def test_epoch_end(
+        self, outputs: Union[List[Dict[str, torch.Tensor]], List[List[Dict[str, torch.Tensor]]]]
+    ) -> Optional[Dict[str, Dict[str, torch.Tensor]]]:
+        return self.asr_model.test_epoch_end(outputs=outputs)
+
+    def val_dataloader(self):
+        return self.asr_model.val_dataloader()
+
+    def unfreeze(self) -> None:
+        super().unfreeze()
+        self.tts_model.freeze()  # tts model should be always frozen
+
+    def on_fit_start(self):
```

Comment:
Same. If only to get some people to understand the idea of ​​training asr-tts without looking into the methods' body.

Reply:
Added docstring